polar-core/src/error.rs

```diff
@@ -408,6 +411,15 @@ impl From<OperationalError> for PolarError {
     }
 }
 
+impl From<RuntimeError> for PolarError {
```

Comment:
@gw000 @gj intentionally removed these conversion routines in #1368

Reply:
whyyyyyyyyy this is bondage & discipline stuff 😖

Reply:
Can I please put it back, and you just don't use it if it's wrong? Nothing has a `context` in here and this also stops us from being able to use `?` with `as_whatever()` methods

Reply:
Do you need to use different types of errors in `filter.rs` besides `RuntimeError`? If you're just interacting with `RuntimeError` you should be able to use `Result<..., RuntimeError>` in that module, and keep using `?`.

It was hard to keep track of where we had properly attached context to errors. So the intent behind this change was to use more specific result types (`Result<(), RuntimeError>`) in modules rather than `PolarError`, and to use `PolarError` vs `RuntimeError`, `ParserError` etc., to distinguish between errors that have had context attached or haven't (distinct from whether they have a source or not).

Long term this is pretty limiting because we might not have source info, or we may want to return multiple different types of errors in one module, so a better solution would be to separate out the two concepts (error with context attached / error without):

```
struct WithContext<E> {
    context : ....
    error: E
}
```

Then you can have `WithContext<RuntimeError>` or `WithContext<PolarError>` if you want.

Reply:
Can you do something like this:

```
# polar.rs
    pub fn build_data_filter(
        &self,
        types: Types,
        partial_results: PartialResults,
        variable: &str,
        class_tag: &str,
    ) -> PolarResult<Filter> {
        Filter::build(types, partial_results, variable, class_tag).map_err(|err| err.with_context(self.kb))
    }
```

```
# filter.rs
impl Filter {
    pub fn build(...) -> Result<..., RuntimeError>
```

Then context is attached in `polar.rs` & conversion also happens there?